go/geth_adapter/adapter.go

```diff
@@ -347,9 +350,9 @@ func gethToVMErrors(err error, gas tosca.Gas) (tosca.CallResult, error) {
 	case
 		geth.ErrOutOfGas,
 		geth.ErrCodeStoreOutOfGas,
-		geth.ErrDepth,
 		geth.ErrContractAddressCollision,
 		geth.ErrExecutionReverted,
+		geth.ErrMaxInitCodeSizeExceeded,
```

Comment:
Is there a way to figure out for which of those we should consume all the gas and for which not?

Reply:
We could check the code coverage while running the ethereum tests, if all errors are touched and all tests pass, they are handled according to the ethereum specification.

Reply:
can that be unit tested after the ethereum tests are verified ?

Reply:
I will continue to look into this but I think this PR is ready to be merged.